pkg/internal/controller/controller.go

```diff
@@ -170,8 +174,12 @@ func (c *Controller) Start(stop <-chan struct{}) error {
 		// Launch workers to process resources
 		c.Log.Info("Starting workers", "worker count", c.MaxConcurrentReconciles)
 		for i := 0; i < c.MaxConcurrentReconciles; i++ {
-			// Process work items
-			go wait.Until(c.worker, c.JitterPeriod, stop)
+			// Run a worker thread that just dequeues items, processes them, and marks them done.
+			// It enforces that the reconcileHandler is never invoked concurrently with the same object.
+			go wait.Until(func() {
```

Comment:
This can likely use `wait.UntilWithContext()` now

Reply:
Done!

Reply:
:grimacing: maybe that will need to wait until after context is plumbed through Runnable, looks like test is not happy with the ordering of stopping leader election compared to the workers because of this.